app/services/padlock/verifier.rb

```diff
@@ -0,0 +1,33 @@
+module Padlock
+  class Verifier < ::Callable
+    def initialize(headers = {})
+      @headers = headers
+    end
+
+    def call
+      user
+    end
+
+    private
+
+    attr_reader :headers
+
+    def user
+      @user ||= User.find(decoded_auth_token[:user_id]) if decoded_auth_token
+    end
+
+    def decoded_auth_token
+      @decoded_auth_token ||= JwToken::Decrypt.call(http_auth_token)
+    end
+
+    def http_auth_token
+      return unless auth_header
+
+      headers['X-Access-Token'].split(' ').last
+    end
+
+    def auth_header
```

Comment:
you can skip this and use
``` 
headers['X-Access-Token'].to_s.split(' ').last
```
in above method.

anyway decryptor would return you nil in `decoded_auth_token`